pkg/asset/installconfig/kubevirt/client.go

```diff
@@ -0,0 +1,255 @@
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	unstructured "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	kubevirtapiv1 "kubevirt.io/client-go/api/v1"
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+//go:generate mockgen -source=./client.go -destination=./mock/client_generated.go -package=mock
+
+// ClientBuilderFuncType is function type for building infra-cluster clients
+type ClientBuilderFuncType func() (Client, error)
+
+// Client is a wrapper object for actual infra-cluster clients: kubernetes and the kubevirt
+type Client interface {
+	GetVirtualMachine(namespace string, name string) (*kubevirtapiv1.VirtualMachine, error)
+	ListVirtualMachine(namespace string) (*kubevirtapiv1.VirtualMachineList, error)
+	DeleteVirtualMachine(namespace string, name string) error
+	GetDataVolume(namespace string, name string) (*cdiv1.DataVolume, error)
+	ListDataVolume(namespace string) (*cdiv1.DataVolumeList, error)
+	DeleteDataVolume(namespace string, name string) error
+	GetSecret(namespace string, name string) (*corev1.Secret, error)
+	ListSecret(namespace string) (*corev1.SecretList, error)
+	DeleteSecret(namespace string, name string) error
+	GetStorageClass(ctx context.Context, name string) (*storagev1.StorageClass, error)
+	GetNetworkAttachmentDefinition(ctx context.Context, name string, namespace string) (*unstructured.Unstructured, error)
+}
+
+type client struct {
+	kubernetesClient *kubernetes.Clientset
+	dynamicClient    dynamic.Interface
+}
+
+// NewClient creates our client wrapper object for the actual kubeVirt and kubernetes clients we use.
+func NewClient() (Client, error) {
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	// if you want to change the loading rules (which files in which order), you can do so here
+
+	configOverrides := &clientcmd.ConfigOverrides{}
+	// if you want to change override values or bind them to flags, there are methods to help you
+
+	kubeConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
+	restClientConfig, err := kubeConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	result := &client{}
+
+	if result.kubernetesClient, err = kubernetes.NewForConfig(restClientConfig); err != nil {
+		return nil, err
+	}
+	if result.dynamicClient, err = dynamic.NewForConfig(restClientConfig); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// VirtualMachine
+
+func (c *client) GetVirtualMachine(namespace string, name string) (*kubevirtapiv1.VirtualMachine, error) {
+	resp, err := c.getResource(namespace, name, vmRes())
```

Comment:
must we use the dynamic client because of colliding deps?

Reply:
Yes, KubeVirt is using k8s mudules version v0.16.4 while the installer is using version v0.19.0